src/unix/linux-inotify.c

```diff
@@ -108,6 +108,40 @@ static int init_inotify(uv_loop_t* loop) {
 }
 
 
+int uv__inotify_fork(uv_loop_t* loop, void* old_watchers) {
+  /* open the inotify_fd, and re-arm all the inotify watchers */
+  int err;
+  struct watcher_list* tmp_watcher_list;
+  struct watcher_list* watcher_list;
+  QUEUE* q;
+  uv_fs_event_t* handle;
+  char* path;
+
+  if (old_watchers) {
+    /* We must restore the old watcher list to be able to close items
+     *  out of it.
+     */
+    loop->inotify_watchers = old_watchers;
+
+    err = init_inotify(loop);
+    if (err)
+      return err;
+
+    RB_FOREACH_SAFE(watcher_list, watcher_root, CAST(&old_watchers), tmp_watcher_list) {
+      path = watcher_list->path;
+      QUEUE_FOREACH(q, &watcher_list->watchers) {
+        handle = QUEUE_DATA(q, uv_fs_event_t, watchers);
+        uv_fs_event_stop(handle);
+        uv_fs_event_start(handle, handle->cb, path, 0);
```

Comment:
Yeah, that comes from my misreading of how it was initially allocated.

e32e1cc4 should handle that better, as well as the iteration issues. (I cargo-culted the "QUEUE_MOVE trick" from later on in the file, so I hope it's correct...)